### PR TITLE
Fix layer opacity is ignored when exporting layouts for layers which are involved in label masks

### PR DESCRIPTION
**Pull request by nyalldawson**

### CODE REVIEW

**Comment by nyalldawson** [comment]
I'll wait till https://github.com/qgis/QGIS/pull/50786 is merged before adding a test to this one.

**Comment by nyalldawson** [comment]
@troopa81 can i get your eyes on this please?

File: tests/src/python/test_selective_masking.py [hunk 1]
```diff
@@ -1329,6 +1330,100 @@ def test_vector_random_generator_fill(self):
...
+        p.end()
+
+        self.assertTrue(self.imageCheck('layer_opacity_points', 'layer_opacity_points', im))
+
+    def imageCheck(self, name, reference_image, image):
+        TestSelectiveMasking.report += f"<h2>Render {name}</h2>\n"
+        temp_dir = QDir.tempPath() + '/'
+        file_name = temp_dir + 'plot_' + name + ".png"
+        image.save(file_name, "PNG")
+        checker = QgsRenderChecker()
```
Comment (troopa81) [inline tests/src/python/test_selective_masking.py:1417]: there is already a checker initialized in setUp, is there a reason to not use it?

File: tests/src/python/test_selective_masking.py [hunk 2]
```diff
@@ -1329,6 +1330,100 @@ def test_vector_random_generator_fill(self):
...
+        checker.setControlPathPrefix("selective_masking")
+        checker.setControlName("expected_" + reference_image)
+        checker.setRenderedImage(file_name)
+        checker.setColorTolerance(2)
+        result = checker.compareImages(name, 20)
+        TestSelectiveMasking.report += checker.report()
```
Comment (troopa81) [inline tests/src/python/test_selective_masking.py:1423]:
```suggestion
        self.report += checker.report()
```

File: tests/src/python/test_selective_masking.py [hunk 3]
```diff
@@ -1329,6 +1330,100 @@ def test_vector_random_generator_fill(self):
...
+
+        job = QgsMapRendererCustomPainterJob(map_settings, p)
+        job.start()
+        job.waitForFinished()
+
+        p.end()
+
+        self.assertTrue(self.imageCheck('layer_opacity', 'layer_opacity', im))
+
+        polys.setOpacity(1.0)
```
Comment (troopa81) [inline tests/src/python/test_selective_masking.py:1395]: In the resulting expected [image](https://github.com/qgis/QGIS/pull/51995/files#diff-0b8924dfd96c55d5ee160cf1543fcf78059a04631b3b004544717515c6d5b00a) points are not masked anymore, although they should be no?

**Review by troopa81 — APPROVED** [review]
Modifications look good

**Pull request closed without merging**